pxemgr/ipxe_handlers.go

```diff
@@ -65,56 +65,33 @@ func (mgr *pxeManagerT) maybeCreateHost(serial string) (*hostmgr.Host, error) {
 
 		if host.InternalAddr == nil {
 			host.InternalAddr = mgr.getNextInternalIP()
-			err = host.Commit("updated host InternalAddr")
-			if err != nil {
-				return nil, microerror.Mask(err)
-			}
 		}
-
+		// generate addresses for the extra NICs
+		extraNicCount := len(mgr.config.Network.ExtraNICs)
+		host.AdditionalAddrs = make(map[string]net.IP)
+		for i := 0; i < extraNicCount; i++ {
+			host.AdditionalAddrs[mgr.config.Network.ExtraNICs[i].InterfaceName] = mgr.getNextAdditionalIP(i, mgr.config.Network.ExtraNICs[i].InterfaceName)
+		}
```

Comment:
assign additional IP to ExtraNICs when new machine is added to the configuration